src/components/ModelsPage/ModelEndpoints/modelEndpoints.util.js

```diff
@@ -54,12 +55,32 @@ const detailsMenu = [
   }
 ]
 
-export const generatePageData = () => ({
+export const generatePageData = (
+  selectedItem,
+  model_monitoring_dashboard_url,
+  handleMonitoring
+) => ({
   page: MODELS_PAGE,
   hidePageActionMenu: true,
   details: {
     menu: detailsMenu,
     infoHeaders,
-    type: MODEL_ENDPOINTS_TAB
+    type: MODEL_ENDPOINTS_TAB,
+    actionButton: {
+      label: 'Resource monitoring',
+      tooltip: !model_monitoring_dashboard_url ? 'Grafana service unavailable' : '',
+      variant: TERTIARY_BUTTON,
+      disabled: !model_monitoring_dashboard_url,
+      onClick: () => handleMonitoring(selectedItem)
+    }
   }
 })
+
+export const monitorModelEndpoint = (model_monitoring_dashboard_url, item, projectName) => {
+  console.log('item', item)
+  let redirectUrl = model_monitoring_dashboard_url
+    .replace('{project}', projectName)
+    .replace('{model_endpoint}', item.metadata?.uid)
```

Comment:
what if the item doesn't have a uid, where should it redirect?

Reply:
@mariana-furyk , According to @assaf758 there must uid - it's the key of endpoint - it's a bug if it's empty.